packages/medusa-payment-stripe/src/core/stripe-base.ts

```diff
@@ -0,0 +1,327 @@
+      intentRequest.automatic_payment_methods = { enabled: true }
+    }
+
+    if (customer?.metadata?.stripe_id) {
+      intentRequest.customer = customer.metadata.stripe_id as string
+    } else {
+      let stripeCustomer
+      try {
+        stripeCustomer = await this.stripe_.customers.create({
+          email,
+        })
+      } catch (e) {
+        return this.buildError(
+          "An error occurred in InitiatePayment during the creation of the stripe customer",
+          e
+        )
+      }
+
+      intentRequest.customer = stripeCustomer.id
+    }
+
+    let session_data
+    try {
+      session_data = (await this.stripe_.paymentIntents.create(
+        intentRequest
+      )) as unknown as Record<string, unknown>
+    } catch (e) {
+      return this.buildError(
+        "An error occurred in InitiatePayment during the creation of the stripe payment intent",
+        e
+      )
+    }
+
+    return {
+      session_data,
+      update_requests: customer?.metadata?.stripe_id
+        ? undefined
+        : {
+            customer_metadata: {
+              stripe_id: intentRequest.customer,
+            },
+          },
+    }
+  }
+
+  async authorizePayment(
+    paymentSessionData: Record<string, unknown>,
+    context: Record<string, unknown>
+  ): Promise<
+    | PaymentProcessorError
+    | {
+        status: PaymentSessionStatus
+        data: PaymentProcessorSessionResponse["session_data"]
+      }
+  > {
+    const status = await this.getPaymentStatus(paymentSessionData)
+    return { data: paymentSessionData, status }
+  }
+
+  async cancelPayment(
+    paymentSessionData: Record<string, unknown>
+  ): Promise<
+    PaymentProcessorError | PaymentProcessorSessionResponse["session_data"]
+  > {
+    try {
+      const id = paymentSessionData.id as string
+      return (await this.stripe_.paymentIntents.cancel(
+        id
+      )) as unknown as PaymentProcessorSessionResponse["session_data"]
+    } catch (error) {
+      if (error.payment_intent?.status === ErrorIntentStatus.CANCELED) {
+        return error.payment_intent
+      }
+
+      return this.buildError(
+        "An error occurred in cancelPayment during the cancellation of the payment",
+        error
+      )
+    }
+  }
+
+  async capturePayment(
+    context: PaymentProcessorContext
+  ): Promise<
+    PaymentProcessorError | PaymentProcessorSessionResponse["session_data"]
+  > {
+    const id = context.paymentSessionData.id as string
+    try {
+      const intent = await this.stripe_.paymentIntents.capture(id)
+      return intent as unknown as PaymentProcessorSessionResponse["session_data"]
+    } catch (error) {
+      if (error.code === ErrorCodes.PAYMENT_INTENT_UNEXPECTED_STATE) {
+        if (error.payment_intent?.status === ErrorIntentStatus.SUCCEEDED) {
+          return error.payment_intent
+        }
+      }
+
+      return this.buildError(
+        "An error occurred in deletePayment during the capture of the payment",
+        error
```

Comment:
todo: Simplify error message
```suggestion
        "An error occurred in deletePayment",
        error
```